core/src/test/kotlin/org/rewedigital/katana/SetTests.kt

```diff
@@ -186,5 +186,26 @@ object SetTests : Spek(
                 set shouldHaveSize 1
                 set shouldContain "Hello world"
             }
+
+            it("should locate dependencies declared in outer scope via get") {
+                val module = Module {
+                    factory(name = "outer scope") { "Hello world" }
+                    singleton { "Another string" }
+                    factory<CharSequence> { "A char sequence" }
+
+                    set<CharSequence> {
+                        get<CharSequence, String>(name = "outer scope")
+                        get<CharSequence, String>()
```

Comment:
Why is `CharSequence` needed if the set is that type already and there can't be another type there?

Reply:
Unfortunately if the type of the set and the type of the requested dependency differ, both types must be specified. In the third example of the test you can see that both types can be omitted if they match. This is an issue of Kotlin's type inference. I hope that the compiler improves in the future so that at least the first type (`CharSequence`) can be inferred.

Reply:
I guess there's no choice, it seems like Kotlin doesn't infer this...